### PR TITLE
[ENH] Implement CumSumTransformer

### LINKED ISSUE #10157: [ENH] Add CumSumTransformer for cumulative sum of time series
**Is your feature request related to a problem? Please describe.**
There are scenarios where the cumulative sum of a time series needs to be calculated within a pipeline (e.g., transforming differences back to cumulative values, or feature engineering).

**Describe the solution you'd like**
I'd like to add a `CumSumTransformer` to `sktime.transformations.series` that calculates the cumulative sum along the time axis, acting as a wrapper around `numpy.cumsum()`. It should also support `inverse_transform` (using `numpy.diff`).

**Additional context**
I would like to contribute this for ESoC 2026.

### PR DESCRIPTION
**Pull request by Vaishnav88sk**
#### Reference Issues/PRs

Resolves #10157 

#### What does this implement/fix? Explain your changes.
This PR introduces the `CumSumTransformer` to `sktime.transformations.series`. It computes the cumulative sum of the input time series and provides the `inverse_transform` capability by taking the discrete difference.

#### Does your contribution introduce a new dependency? Please describe.
No.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by jgyasu** [comment]
This account has opened 5 issues and 5 PRs in a span of 10-15 minutes. I highly suspect this is an agent.

**Comment by Vaishnav88sk** [comment]
Hello @jgyasu sir, not an agent or a bot. I had been working on some code changes locally to address the issues, but due to my workload, I wasn’t able to push them earlier. My intention was to complete and merge the contribution from my side, which is why I followed the proper process of raising an issue first and then submitting the associated PR. (Sorry for not waiting for the issue assign. ;) )
I’m currently applying for eSOC and have some open-source contribution experience. I realize that pushing everything together was a result of my own mismanagement, and I sincerely apologize for that. I submitted the changes within a short span (around an hour) as it took me some time to properly write and organize the PR content. I apologize again for the mismanagement, and I would like to clarify that the contributions are entirely my own work.
Further more I have 2-3 more issues/enhancements in my mind for the repo and will propose that in next 2 days.
I would really appreciate your review and feedback on the changes. I had also reached out earlier via email regarding contributions and a few questions about the project.
Thank you for your time and guidance.

**Comment by Vaishnav88sk** [comment]
### What this achieves
This PR introduces the `CumSumTransformer` to `sktime.transformations.series`, wrapping `numpy.cumsum`. It computes the cumulative sum of a time series over time. Crucially, it also implements `inverse_transform` using `numpy.diff`.

### Real-world use case
Often, data is provided in differences or flows (e.g., daily new users, daily sales, or hourly rainfall), but a specific forecasting model might perform better when modeling the cumulative stock (e.g., total active users, cumulative monthly sales). 

By having this as a native transformer with an `inverse_transform`, users can wrap forecasters that model the cumulative series and automatically inverse-transform the predictions back to the original differenced scale.

### Example Usage
```python
from sktime.transformations.series.cumsum import CumSumTransformer
from sktime.datasets import load_airline

y_daily_additions = load_airline() 

transformer = CumSumTransformer()
y_cumulative = transformer.fit_transform(y_daily_additions)

# The inverse returns it to the original differenced scale
y_original = transformer.inverse_transform(y_cumulative)
```
While using this for my college work, I thought this could be easier. So this PR. Thank you!

**Comment by Vaishnav88sk** [comment]
@jgyasu I would like your review on this. And if it is ready to merge?
